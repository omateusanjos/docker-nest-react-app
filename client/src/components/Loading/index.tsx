import React from "react";
import ReactLoading, { LoadingType } from "react-loading";

const Loading = ({
  type,
  color,
  ...rest
}: {
  type: LoadingType;
  color: string;
}) => (
  <ReactLoading
    type={type}
    color={color}
    height={"20%"}
    width={"20%"}
    delay={2}
    {...rest}
  />
);

export default Loading;
