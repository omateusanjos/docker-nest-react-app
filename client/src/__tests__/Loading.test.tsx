import React from "react";
import Loading from "../components/Loading";
import { render } from "@testing-library/react";

describe("\nLoading component", () => {
  it("should render a loading component", () => {
    const { getByTestId, container } = render(
      <Loading type="cubes" color="#000000" data-testid="loading" />
    );
    expect(getByTestId("loading")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("should render a loading component with a different type", () => {
    const { getByTestId, rerender } = render(
      <Loading type="cubes" color="#000000" data-testid="loading" />
    );
    expect(getByTestId("loading")).toBeInTheDocument();

    rerender(
      <Loading
        type="spinningBubbles"
        color="#000000"
        data-testid="loading-new-type"
      />
    );
    expect(getByTestId("loading-new-type")).toBeInTheDocument();
  });

  it("should render a loading component with a different color", () => {
    const { getByTestId, rerender } = render(
      <Loading type="cubes" color="#ffffff" data-testid="loading" />
    );
    expect(getByTestId("loading")).toBeInTheDocument();
    expect(getByTestId("loading")).toHaveStyle("fill: #ffffff");

    rerender(<Loading type="cubes" color="#000000" data-testid="loading" />);
    expect(getByTestId("loading")).toBeInTheDocument();
    expect(getByTestId("loading")).toHaveStyle("fill: #000000");
  });
});
