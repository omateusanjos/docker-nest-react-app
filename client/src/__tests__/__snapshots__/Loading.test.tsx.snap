// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`
Loading component should render a loading component 1`] = `
<div>
  <div
    data-testid="loading"
    style="fill: #000000; height: 20%; width: 20%;"
  >
    <svg
      class="icon-blank"
      viewBox="0 0 32 32"
      xmlns="http://www.w3.org/2000/svg"
    />
    

  </div>
</div>
`;
